tests/test_split_batch.py

```diff
@@ -85,46 +77,84 @@ def dummy_batches(batch_size=12):
         dummy_tensor_batch(batch_size=batch_size),
         dummy_tuple_batch(batch_size=batch_size),
         dummy_tuple_batch_long(batch_size=batch_size),
-        dummy_tuple_list_batch(batch_size=batch_size),
         dummy_dict_batch(batch_size=batch_size),
         dummy_dict_batch_with_metadata(batch_size=batch_size)
     ]
 
 
 @pytest.mark.parametrize('batch', dummy_batches(12))
 def test_split_without_error(batch):
-    _default_split_batch(batch, num_microbatches=3)
+    microbatches = _default_split_batch(batch, microbatch_size=3)
+    assert len(microbatches) == 4
 
 
 @pytest.mark.parametrize('batch', [dummy_tuple_batch(12)])
 def test_split_tuple(batch):
-    microbatches = _default_split_batch(batch, num_microbatches=3)
+    microbatches = _default_split_batch(batch, microbatch_size=4)
     # should be 3 microbatches of size 4 tensors pairs
     # should split into [(x, y), (x, y), (x, y)]
     assert len(microbatches[0]) == 2
 
 
 @pytest.mark.parametrize('batch', [dummy_tuple_batch_long(12)])
 def test_split_tuple_long(batch):
-    microbatches = _default_split_batch(batch, num_microbatches=3)
+    microbatches = _default_split_batch(batch, microbatch_size=4)
     assert len(microbatches[0]) == 4
 
 
+@pytest.mark.parametrize('batch', dummy_batches(5))
+def test_odd_batch_sizes(batch):
+    microbatches = _default_split_batch(batch, microbatch_size=2)
+    # should split into [len(2), len(2), len(2)]
+    assert len(microbatches) == 3
+    last_microbatch = microbatches[-1]
+    if isinstance(last_microbatch, Mapping):
+        assert len(last_microbatch['image']) == 1
+        assert len(last_microbatch['target']) == 1
+    if isinstance(last_microbatch, tuple):
+        assert len(last_microbatch[0]) == 1
+    if isinstance(last_microbatch, list):
+        assert len(last_microbatch) == 1
+
+
+@pytest.mark.parametrize('batch', dummy_batches(1))
+def test_batch_size_less_than_num_microbatches(batch):
+    with pytest.raises(ValueError):
+        _default_split_batch(batch, microbatch_size=3)
+
+
 @pytest.mark.parametrize('batch', [dummy_maskrcnn_batch(12)])
-def test_split_maskrcnn(batch):
-    microbatches = _split_list(batch, num_microbatches=3)
+def test_num_microbatches_split_maskrcnn(batch):
```

Comment:
I think this test should be below the comment about older tests?

Reply:
Since it calls an underlying function below grad accum / microbatch size abstraction, I'm going to keep it

Reply:
oh woops I guess i just looked at the name not the implementation. so shouldn't it be called `test_microbatch_size_split_maskrcnn`?

Reply:
oh good catch, forgot to rename